app/helpers/application_helper.rb

```diff
@@ -408,14 +408,19 @@ def topic_featured_link_domain(link)
     end
   end
 
-  def theme_ids
+  def theme_id
     if customization_disabled?
-      [nil]
+      nil
     else
-      request.env[:resolved_theme_ids]
+      request.env[:resolved_theme_id]
     end
   end
 
+  def stylesheet_manager
+    return @stylesheet_manager if defined?(@stylesheet_manager)
+    @stylesheet_manager = Stylesheet::Manager.new(theme_id: theme_id)
```

Comment:
Each request will work with an instance of `Stylesheet::Manager` which stores state of all the themes that has been loaded within a request. This allows us to reuse the state across method calls which wasn't possible previously.